### PR TITLE
Bump Ambassador version to 0.50.3.

### PR DESCRIPTION
**Pull request by kflynn**
Bump Ambassador chart to Ambassador version 0.50.3.

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/ambassador/Chart.yaml
Comment (steven-sheehy) [inline stable/ambassador/Chart.yaml:2]: Need to bump version in values.yaml and in README

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/ambassador/Chart.yaml
Comment (kflynn) [reply stable/ambassador/Chart.yaml:2]: Oh dear. Thanks for the catch!!

File: stable/ambassador/Chart.yaml
Comment (steven-sheehy) [inline stable/ambassador/Chart.yaml:8]: hmm, while you're in there can you add the other person in OWNERS to the list of maintainers?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by Flydiverny** [comment]
Think it makes sense to update maintainers on chart! Also just like to note that when we changed from kflynn to flynn in the maintainers the link on https://hub.helm.sh/charts/stable/ambassador now points wrongly to the github organisation flynn :)

File: stable/ambassador/Chart.yaml
Comment (steven-sheehy) [reply stable/ambassador/Chart.yaml:8]: And change `Flynn` to `kflynn`. It should be your github userid.

**Comment by steven-sheehy** [comment]
Yeah, it should be kflynn in Chart.yaml since that's the github userid. I added a comment.

File: stable/ambassador/Chart.yaml
Comment (kflynn) [reply stable/ambassador/Chart.yaml:8]: Updated, thanks!

**Comment by Flydiverny** [comment]
@kflynn missing sign-off :)

**Comment by kflynn** [comment]
Oh for crying out loud... 😆  I don't suppose I can tell the bot that no, really, the fact that my commits are GPG-signed is good enough??? 😄

**Comment by kflynn** [comment]
(Thanks for the sharp eyes, @Flydiverny and @steven-sheehy. 😄 )

**Comment by Flydiverny** [comment]
😂 
/lgtm

**Pull request merged**
